libs/domains/service-helm/feature/src/lib/networking-port-setting-modal/networking-port-setting-modal.tsx

```diff
@@ -0,0 +1,184 @@
+import { type HelmPortRequestPortsInner, PortProtocolEnum } from 'qovery-typescript-axios'
+import { type FormEvent } from 'react'
+import { Controller, FormProvider, useForm } from 'react-hook-form'
+import { InputSelect, InputText, ModalCrud } from '@qovery/shared/ui'
+
+export interface NetworkingPortSettingModalProps {
+  port?: HelmPortRequestPortsInner
+  onClose: () => void
+  onSubmit: (port: HelmPortRequestPortsInner) => void
+}
+
+export function NetworkingPortSettingModal({ port, onClose, onSubmit }: NetworkingPortSettingModalProps) {
+  const isEdit = !!port
+  const methods = useForm({
+    defaultValues: {
+      internal_port: port?.internal_port ?? '',
+      external_port: port?.external_port ?? '443',
+      service_name: port?.service_name ?? '',
+      namespace: port?.namespace ?? '',
+      protocol: port ? port.protocol : PortProtocolEnum.HTTP,
+      name: port ? port.name : '',
+    },
+    mode: 'onChange',
+  })
+
+  const { control, watch, setValue } = methods
+
+  const pattern = {
```

Comment:
can be directly on your controller you use this pattern only one time

Reply:
Fixed